byterun/domain.c

```diff
@@ -642,6 +642,12 @@ int caml_try_run_on_all_domains_with_spin_work(
   int i;
   uintnat domains_participating = 0;
 
+  // Don't take the lock if there's already a stw leader
+  if( stw_leader ) {
```

Comment:
Just to be consistent with the use of `stw_leader` in the rest of the code, should we use an atomic_load on this?